src/GHCMod.hs

```diff
@@ -256,6 +257,14 @@ globalArgSpec =
                reqArg "OPT" $ \g o ->
                    o { ghcUserOptions = g : ghcUserOptions o }
 
+      , option "" ["map-file"] "Redirect one file to another, --map-file \"file1.hs=file2.hs\"" $
```

Comment:
Could you add a more detailed user/developer facing description here as a comment, I'll expose it in the `--help` output as part of moving to optparse-applicative later.